include/sentry.h

```diff
@@ -1286,6 +1286,32 @@ SENTRY_EXPERIMENTAL_API sentry_value_t sentry_start_transaction(
 SENTRY_EXPERIMENTAL_API void sentry_transaction_finish(
     sentry_value_t transaction);
 
+/**
+ * Sets a tag on a transaction to the given string value.
+ */
+SENTRY_EXPERIMENTAL_API void sentry_transaction_set_tag(
+    sentry_value_t transaction, const char *tag, const char *value);
+
+/**
+ * Removes a tag from a transaction. Returns 0 if the tag was successfully
+ * removed, 1 otherwise.
+ */
+SENTRY_EXPERIMENTAL_API int sentry_transaction_remove_tag(
+    sentry_value_t transaction, const char *tag);
+
+/**
+ * Sets the given key in a transaction's "data" section to the given value.
+ */
+SENTRY_EXPERIMENTAL_API void sentry_transaction_set_data(
+    sentry_value_t transaction, const char *key, sentry_value_t value);
+
+/**
+ * Removes a key from a transaction's "data" section. Returns 0 if the key was successfully
+ * removed, 1 otherwise.
+ */
```

Comment:
is there a particular use case you had in mind and intended to support by returning 1 if the removal of the key fails?

Reply:
No, the idea was just to align the function's signature with that of `remove_by_key`.

Reply:
there is some precedence in ignoring the return value (see: `sentry_remove_tag`, `sentry_remove_extra`, `sentry_remove_user`, etc.) so i would personally prefer to align with that. it isn't clear what needs to be done if the key is unsuccessfully removed, which leads me to think that it may be more useful to just return `void`. what do you do if you get `1`? do you just call `remove_by_key` again? ignore the return value?

to compound on that, you could argue that the two scenarios where `1` is returned are cases where the key is "successfully" removed in that it no longer exists in the `sentry_value_t`: it wasn't a type that could contain keys that could be removed to begin with, or the key wasn't there to begin with.

Reply:
> there is some precedence in ignoring the return value (see: sentry_remove_tag, sentry_remove_extra, sentry_remove_user, etc.) 

Shoot, I didn't see those. You're right that the new methods should keep the same signature.